src/compiler/checker.ts

```diff
@@ -9569,16 +9878,24 @@ module ts {
                     forEach(potentialThisCollisions, checkIfThisIsCapturedInEnclosingScope);
                     potentialThisCollisions.length = 0;
                 }
-
+                if (potentialArgumentsCollisions.length) {
+                    forEach(potentialArgumentsCollisions, checkIfArgumentsIsCapturedInEnclosingScope);
+                    potentialArgumentsCollisions.length = 0;
+                }
                 if (emitExtends) {

```

Comment:
That seems inconsistent with: https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24614211